### PR TITLE
Support `optout` patient engagement status

### PR DESCRIPTION
**Pull request by LisaManresa**
- [x] Add the Clubhouse Story ID: [ch23008]

### CODE REVIEW

File: src/js/i18n/en-US.yml [hunk 1]
```diff
@@ -481,6 +481,9 @@ careOptsFrontend:
                 not_responding {Not Responding}
                 paused {Paused}
                 scheduled {Scheduled}
+                optout {Opted Out}
+                bad_email {Bad Email}
```
Comment (paulfalgout) [inline src/js/i18n/en-US.yml]: @luketlancaster `bad_email` `created`?

Comment (luketlancaster) [reply src/js/i18n/en-US.yml]: the only use I can find for `bad_email` in the old code is for login errors for clinician users...not sure why it was in the i18n for engagement that I sent to @LisaManresa

Comment (luketlancaster) [reply src/js/i18n/en-US.yml]: also not sure about `created`....old value that's not used anymore maybe? Here's what the old system has: https://github.com/RoundingWell/RWell-rwmicro/blob/master/shared/i18n/en-US.yml#L407-L417

Comment (paulfalgout) [reply src/js/i18n/en-US.yml]: On legacy frontend support [this](https://github.com/RoundingWell/RWell/blob/b666bbb1030c3324aa931a51a8af5775b5893ef8/mainapp/assets/js/static-options.js#L1):
```js
export const PATIENTSTATUS = {
    ENGAGEMENT_OFF: 'disabled',
    NOT_SCHEDULED: 'enrolled',
    SCHEDULED: 'scheduled',
    ACTIVE: 'active',
    NOT_RESPONDING: 'not_responding',
    INACTIVE: 'inactive',
    PAUSED: 'paused',
    OPTOUT: 'optout'
};
```
so I don't think the others are in use?  Also what about off/disabled?
@luketlancaster

Comment (luketlancaster) [reply src/js/i18n/en-US.yml]: `'disabled'` is it's own status that comes with an `engagement_off_reason`, which is a string of various values (clinically na, other, no access, etc)

and looks like status is going to be `'disabled'` if engagement is off/on initial load-in of patients to the old system


does that answer any of your questions?

Comment (paulfalgout) [reply src/js/i18n/en-US.yml]: how does a patient get disabled?... or paused for that matter?  I understand we have the `engagement_off_reason` things but to my knowledge the only way to set that is through the legacy interface.

Comment (luketlancaster) [reply src/js/i18n/en-US.yml]: so I think the only way for engagement to be disabled, currently, is for either them to stop responding and so eventually the system turns it off, or for them to opt-out in the surveyapp. If they opt out via surveyapp they get a status of `'optout'`.

Comment (paulfalgout) [reply src/js/i18n/en-US.yml]: I don't think patients go off by themselves.. they go inactive.. _pretty_ sure.. I mean if demo is still working there are tons of inactive patients.

**Review by paulfalgout — APPROVED** [review]

**Pull request merged**